src/Pelagos/Bundle/AppBundle/Controller/UI/ExternalDownloadLogController.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Controller\UI;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+use Symfony\Component\Form\Form;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+use Pelagos\Bundle\AppBundle\Form\ExternalDownloadLogType;
+
+use Pelagos\Entity\Account;
+use Pelagos\Entity\Dataset;
+
+/**
+ * The end review tool helps to end the review of a dataset submission review.
+ *
+ * @Route("/external-download-log")
+ */
+class ExternalDownloadLogController extends UIController
+{
+
+    /**
+     * The default action for End Review.
+     *
+     * @param Request $request The Symfony request object.
+     *
+     * @Route("")
+     *
+     * @return Response A Response instance.
+     */
+    public function defaultAction(Request $request)
+    {
+        if (!$this->isGranted('ROLE_DATA_REPOSITORY_MANAGER')) {
+            return $this->render('PelagosAppBundle:template:AdminOnly.html.twig');
+        }
+
+        $form = $this->get('form.factory')->createNamed(
+            'externalDownloadLog',
+            ExternalDownloadLogType::class
+        );
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $formData = $this->getFormData($form);
+            $udi = substr($formData['udi'], 0, 16);
+            $typeId = '';
+            $type = '';
+            $datasets = $this->entityHandler->getby(Dataset::class, array('udi' => $udi));
+            $dataset = $datasets[0];
+            if ($dataset instanceof Dataset) {
+                if ($formData['userType']) {
+                    $account = $this->entityHandler
+                        ->getBy(Account::class, array('userId' => $formData['username']));
+                    if ($account[0]) {
```

Comment:
Check if there is even an element 0, then also check if instanceof account. (i guess)

Reply:
even better, throw an exception here, saying, user not found or something.

Reply:
Added try-catch block around it 👍